test/cpp/qps/client_async.cc

```diff
@@ -130,39 +130,26 @@ class ClientRpcContextUnaryImpl : public ClientRpcContext {
       response_reader_;
 };
 
-class AsyncUnaryClient GRPC_FINAL : public Client {
+class AsyncClient : public Client {
  public:
-  explicit AsyncUnaryClient(const ClientConfig& config) : Client(config) {
+  explicit AsyncClient(const ClientConfig& config,
+                       void (*setup_ctx)(CompletionQueue*, TestService::Stub*,
+                                         const SimpleRequest&)) :
+      Client(config) {
     for (int i = 0; i < config.async_client_threads(); i++) {
       cli_cqs_.emplace_back(new CompletionQueue);
     }
-
-    auto check_done = [](grpc::Status s, SimpleResponse* response) {};
-
     int t = 0;
     for (int i = 0; i < config.outstanding_rpcs_per_channel(); i++) {
       for (auto channel = channels_.begin(); channel != channels_.end();
```

Comment:
Every time you add a C++11 feature that breaks crappy compilers, God kills a kitten. Unfortunately, we mostly use constructs like this throughout the gRPC C++ code base rather than more modern features like range-based fors because we support GCC down to 4.4 and VS down to 2010.